### ISSUE TITLE
Generator and Dataset doesn't respect base_option '--output_nc'

### ISSUE THREAD

**Issue by ychnh** [NVlabs/SPADE#85]
When output_nc is set to a value other than 3, (1 for example)
The models and dataloaders are not adjusted accordingly

**data/pix2pix_dataset.py - line 74**
image = image.convert('RGB')
... 
Along with some other PIL image conversions that a problematic for 1channel image

**models/networks/generator.py - line 55**
self.conv_img = nn.Conv2d(final_nc, 3, 3, padding=1)
